src/main/java/org/spongepowered/common/mixin/core/server/MixinServerConfigurationManager.java

```diff
@@ -24,48 +24,367 @@
+        this.setPlayerGameTypeBasedOnOther(playerIn, null, worldserver);
+
+        // Support vanilla clients logging into custom dimensions
+        int dimension = DimensionManager.getClientDimensionToSend(worldserver.provider.getDimensionId(), worldserver, playerIn);
+        if (((IMixinEntityPlayerMP) playerIn).usesCustomClient()) {
+            DimensionManager.sendDimensionRegistration(worldserver, playerIn, dimension);
+        }
+
+        handler.sendPacket(new S01PacketJoinGame(playerIn.getEntityId(), playerIn.theItemInWorldManager.getGameType(), worldinfo
+                .isHardcoreModeEnabled(), dimension, worldserver.getDifficulty(), this.getMaxPlayers(), worldinfo
+                .getTerrainType(), worldserver.getGameRules().getGameRuleBooleanValue("reducedDebugInfo")));
+        handler.sendPacket(new S3FPacketCustomPayload("MC|Brand", (new PacketBuffer(Unpooled.buffer())).writeString(this
+                .getServerInstance().getServerModName())));
+        handler.sendPacket(new S41PacketServerDifficulty(worldinfo.getDifficulty(), worldinfo.isDifficultyLocked()));
+        handler.sendPacket(new S05PacketSpawnPosition(blockpos));
+        handler.sendPacket(new S39PacketPlayerAbilities(playerIn.capabilities));
+        handler.sendPacket(new S09PacketHeldItemChange(playerIn.inventory.currentItem));
+        playerIn.getStatFile().func_150877_d();
+        playerIn.getStatFile().func_150884_b(playerIn);
+        this.mcServer.refreshStatusNextTick();
+
+        this.playerLoggedIn(playerIn);
+        handler.setPlayerLocation(playerIn.posX, playerIn.posY, playerIn.posZ, playerIn.rotationYaw, playerIn.rotationPitch);
+        this.updateTimeAndWeatherForPlayer(playerIn, worldserver);
+
+        if (this.mcServer.getResourcePackUrl().length() > 0) {
+            playerIn.loadResourcePack(this.mcServer.getResourcePackUrl(), this.mcServer.getResourcePackHash());
+        }
+
+        playerIn.addSelfToInternalCraftingInventory();
+
+
+        // Sponge Start
+
+        // Move logic for creating join message up here
+        ChatComponentTranslation chatcomponenttranslation;
+
+        if (!playerIn.getCommandSenderName().equalsIgnoreCase(s))
+        {
+            chatcomponenttranslation = new ChatComponentTranslation("multiplayer.player.joined.renamed", new Object[] {playerIn.getDisplayName(), s});
+        }
+        else
+        {
+            chatcomponenttranslation = new ChatComponentTranslation("multiplayer.player.joined", new Object[] {playerIn.getDisplayName()});
+        }
+
+        chatcomponenttranslation.getChatStyle().setColor(EnumChatFormatting.YELLOW);
+
+        // Fire PlayerJoinEvent
+        final PlayerJoinEvent event = SpongeImplEventFactory.createPlayerJoin(Sponge.getGame(), (Player) playerIn, ((Player) playerIn).getLocation(),
+                SpongeTexts.toText(chatcomponenttranslation), ((Player) playerIn).getMessageSink());
+        Sponge.getGame().getEventManager().post(event);
```

Comment:
Pretty sure this won't post to Forge's event bus. Will need to look into that in `SpongeModEventManager`